File: backend/src/main/java/unpsjb/labprog/backend/model/ArticuloLicencia.java
package unpsjb.labprog.backend.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.SequenceGenerator;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * Entidad que representa un artículo de licencia del reglamento docente. Define
 * los tipos de licencias disponibles con su código y descripción.
 *
 * @author Mariano Bonansea
 * @version 1.0
 */
@Entity
@Table(name = "articulos_licencia")
@Getter
@Setter
@NoArgsConstructor
public class ArticuloLicencia {

    /**
     * Identificador único del artículo de licencia.
     */
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "articulosLicencia_seq_gen")
    @SequenceGenerator(name = "articulosLicencia_seq_gen", sequenceName = "articulosLicencia_seq", initialValue = 1000, allocationSize = 1)
    private int id;

    /**
     * Código del artículo de licencia (ej: "5A", "7B").
     */
    @NotNull
    @Column(length = 10, nullable = false, unique = true)
    private String articulo;

    /**
     * Descripción detallada del tipo de licencia.
     */
    @Column(length = 90)
    private String descripcion;

}

File: frontend/cli/src/app/models/articulo-licencia.ts
/**
 * Interfaz que representa un artículo reglamentario de licencia docente.
 * 
 * Define los tipos de licencias disponibles según la normativa aplicable,
 * especificando el marco legal que ampara cada tipo de ausencia docente.
 * Los artículos de licencia son la base normativa para la validación
 * y tramitación de solicitudes de licencia en el sistema académico.
 * 
 * @interface ArticuloLicencia
 * @author Mariano Bonansea
 * @version 1.0
 */
export interface ArticuloLicencia {
    /**
     * Identificador único del artículo de licencia en el sistema.
     * 
     * Clave primaria autoincremental que identifica unívocamente
     * cada tipo de licencia disponible en el marco normativo.
     */
    id?: number;

    /**
     * Código o número del artículo reglamentario.
     * 
     * Identificador textual único que referencia el artículo específico
     * de la normativa que ampara el tipo de licencia. Debe ser único
     * en el sistema para evitar ambigüedades normativas.
     */
    articulo: string;

    /**
     * Descripción detallada del artículo de licencia.
     * 
     * Texto explicativo que detalla las condiciones, alcances y
     * procedimientos específicos del tipo de licencia, facilitando
     * la comprensión de su aplicación práctica.
     */
    descripcion?: string;
}